src/model_instance.rs

```diff
@@ -350,54 +351,76 @@ impl ModelInstance {
             mi.optimizer = Optimizer::Adagrad;
         }
 
-        
-        
         Ok(mi)
     }
 
-/*
-    pub fn new_from_jsonfile(input_filename: &str, vw: &vwmap::VwNamespaceMap) -> Result<ModelInstance, Box<dyn Error>> {
-        let mut mi = ModelInstance::new_empty()?;
-        let mut input = File::open(input_filename)?;
-        let mut contents = String::new();
-        input.read_to_string(&mut contents)?;
-        let j: Value = serde_json::from_str(&contents)?;
-        let descj = &j["desc"];
-        mi.learning_rate = descj["learning_rate"].as_f64().unwrap() as f32;
-        mi.bit_precision = descj["bit_precision"].as_u64().unwrap() as u8;
-        let features = descj["features"].as_array().unwrap();
-        for feature in features {
-            let mut feature_combo_desc = FeatureComboDesc {
-                                feature_indices: Vec::new(),
-                                weight: 1.0,
-                                };
-
-            let fname = feature.as_str().unwrap();
-            let primitive_features = fname.split(",");
-            for primitive_feature_name in primitive_features {
-                let index = match vw.map_name_to_index.get(primitive_feature_name) {
-                    Some(index) => *index,
-                    None => return Err(Box::new(IOError::new(ErrorKind::Other, format!("Unknown feature name in model json: {}", primitive_feature_name))))
-                };
-                let index = mi.get_namespace_id(vw, primitive_feature_name)?;
-
-                feature_combo_desc.feature_indices.push(index);
-            }
-            mi.feature_combo_descs.push(feature_combo_desc);
-//            mi.feature_combos.push(feature_vec);
-        }
 
-        Ok(mi)
-    }
-    */
+	pub fn update_hyperparameters_from_cmd<'a>(cmd_arguments: &clap::ArgMatches<'a>, mi: &mut ModelInstance) -> Result<(), Box<dyn Error>> {
+		/*! A method that enables updating hyperparameters of an existing (pre-loaded) model.
+		Currently limited to the most commonly used hyperparameters. */
+
+
+        if !cmd_arguments.is_present("unlock_hyperparameters") {
```

Comment:
you've just repeated the same issue as before - this is a silent breakage.

people will be using --learning_rate and think it will be used, while it clearly won't be. 
so adding unlock_hyperparameters is worse than not doing anything at all.

looking at all this, my suggestion is to entirely remove unlock_hyperparameters parameter and not change anything else.
(and ignore comment from above)